README.md

```diff
@@ -32,84 +34,229 @@ The red balls represent the position of the drones.
 
 <img src="circle.gif" alt="Circle" width="30%"/>
 
 The yellow balls represent the target position of the drones.
 
+Available in [Numpy](crazy_rl/multi_agent/numpy/circle/circle.py) and [JAX](crazy_rl/multi_agent/jax/circle/circle.py)
+version.
+
 ### Surround
-The drones learn to [surround](crazy_rl/multi_agent/surround/surround.py) a fixed target point.
+The drones learn to surround a fixed target point.
 
 <img src="surround.gif" alt="Surround" width="30%"/>
 
 The yellow ball represents the target the drones have to surround.
 
+Available in [Numpy](crazy_rl/multi_agent/numpy/surround/surround.py) and
+[JAX](crazy_rl/multi_agent/jax/surround/surround.py) version.
+
 ### Escort
-The drones learn to [escort](crazy_rl/multi_agent/escort/escort.py) a target moving straight to one point to another.
+The drones learn to escort a target moving straight to one point to another.
 
 <img src="escort.gif" alt="Escort" width="30%"/>
 
-The yellow ball represents the target the drones have to escort.
+The yellow ball represents the target the drones have to surround.
+
+Available in [Numpy](crazy_rl/multi_agent/numpy/escort/escort.py) and [JAX](crazy_rl/multi_agent/jax/escort/escort.py)
+version.
 
 ### Catch
-The drones learn to [catch](crazy_rl/multi_agent/catch/catch.py) a target trying to escape.
+The drones learn to catch a target trying to escape.
 
 <img src="catch.gif" alt="Catch" width="30%"/>
 
-The yellow ball represents the target the drones have to catch.
+The yellow ball represents the target the drones have to surround.
+
+Available in [Numpy](crazy_rl/multi_agent/numpy/catch/catch.py) and [JAX](crazy_rl/multi_agent/jax/catch/catch.py)
+version.
 
 ## API
 
 ### Training
 I suggest to have a look at [MASAC](https://github.com/ffelten/MASAC) for training the agents.
 
-### Execution
+### Numpy version
+
+Basic version which can be used for training, simulation and the real drones.
+
+Execution :
+```python
+from crazy_rl.multi_agent.numpy.circle.circle import Circle
+
+env: ParallelEnv = Circle(
+    drone_ids=np.array([0, 1]),
+    render_mode="human",    # or real, or None
+    init_flying_pos=np.array([[0, 0, 1], [2, 2, 1]]),
+)
+
+done = False
+while not done:
+    # Execute policy for each agent
+    actions: Dict[str, np.ndarray] = {}
+    for agent_id in env.possible_agents:
+        obs_with_id = torch.Tensor(concat_id(obs[agent_id], agent_id)).to(device)
+        act, _, _ = actor.get_action(obs_with_id.unsqueeze(0)) # YOUR POLICY HERE
+        act = act.detach().cpu().numpy()
+        actions[agent_id] = act.flatten()
+
+    obs, _, _, _, _ = env.step(actions)
+```
+
+You can have a look at the [test_multiagent](learning/exec_multiagent.py) file. The path to the save model MASAC and the mode to "real" has to be set on the main.
+
+### JAX version
+
+This version is specifically optimized for GPU usage and intended for agent training purposes.
+However, simulation and real-world functionalities are not available in this version.
```

Comment:
Moreover, it is not compliant with the PettingZoo API as it heavily relies on functional programming. We sacrified the API compatibility for huge performance gains.